lib/workers/repository/extract/extract-fingerprint-config.ts

```diff
@@ -0,0 +1,85 @@
+import { getManagerConfig, mergeChildConfig } from '../../../config';
+import type {
+  RegexManagerTemplates,
+  RenovateConfig,
+} from '../../../config/types';
+import { getManagerList } from '../../../modules/manager';
+import { validMatchFields } from '../../../modules/manager/regex/utils';
+import type { CustomExtractConfig } from '../../../modules/manager/types';
+import type { WorkerExtractConfig } from '../../types';
+
+export interface FingerprintExtractConfig {
+  managerList: Set<string>;
+  managers: WorkerExtractConfig[];
+}
+
+function getRegexManagerFields(
+  config: WorkerExtractConfig
+): CustomExtractConfig {
+  const regexFields = {} as CustomExtractConfig;
+  for (const field of validMatchFields.map(
+    (f) => `${f}Template` as keyof RegexManagerTemplates
+  )) {
+    if (config[field]) {
+      regexFields[field] = config[field];
+    }
+  }
+
+  return {
+    autoReplaceStringTemplate: config.autoReplaceStringTemplate,
+    matchStrings: config.matchStrings,
+    matchStringsStrategy: config.matchStringsStrategy,
+    ...regexFields,
+  };
+}
+
+function getFilteredManagerConfig(
+  config: WorkerExtractConfig
+): WorkerExtractConfig {
+  return {
+    ...(config.manager === 'regex' ? getRegexManagerFields(config) : {}),
```

Comment:
```suggestion
    ...(config.manager === 'regex' && getRegexManagerFields(config)),
```

we use this pattern a lot 🙃